client_test.go

```diff
@@ -6,106 +6,136 @@ import (
+		}
+		defer s.Shutdown()
+		c, err := net.Dial("udp", addrstr)
+		if err != nil {
+			t.Fatalf("failed to dial: %v", err)
+		}
+		defer c.Close()
+		if !isPacketConn(c) {
+			t.Error("UDP connection should be a packet conn")
+		}
+		if !isPacketConn(struct{ *net.UDPConn }{c.(*net.UDPConn)}) {
+			t.Error("UDP connection (wrapped type) should be a packet conn")
+		}
+	})
 
-	// Unix Seqpacket
-	shutChan, addrstr, err := RunLocalUnixSeqPacketServer(filepath.Join(t.TempDir(), "unixpacket.sock"))
-	if err != nil {
-		t.Fatalf("unable to run test server: %v", err)
-	}
+	t.Run("TCP", func(t *testing.T) {
+		s, addrstr, _, err := RunLocalTCPServer(":0")
+		if err != nil {
+			t.Fatalf("unable to run test server: %v", err)
+		}
+		defer s.Shutdown()
+		c, err := net.Dial("tcp", addrstr)
+		if err != nil {
+			t.Fatalf("failed to dial: %v", err)
+		}
+		defer c.Close()
+		if isPacketConn(c) {
+			t.Error("TCP connection should not be a packet conn")
+		}
+		if isPacketConn(struct{ *net.TCPConn }{c.(*net.TCPConn)}) {
+			t.Error("TCP connection (wrapped type) should not be a packet conn")
+		}
+	})
 
-	defer func() {
-		shutChan <- &struct{}{}
-	}()
-	c, err = net.Dial("unixpacket", addrstr)
-	if err != nil {
-		t.Fatalf("failed to dial: %v", err)
-	}
-	defer c.Close()
-	if !isPacketConn(c) {
-		t.Error("Unix datagram connection should be a packet conn")
-	}
-	if !isPacketConn(struct{ *net.UnixConn }{c.(*net.UnixConn)}) {
-		t.Error("Unix datagram connection (wrapped type) should be a packet conn")
-	}
+	t.Run("Unix datagram", func(t *testing.T) {
+		s, addrstr, _, err := RunLocalUnixGramServer(filepath.Join(tempDir(t), "unixgram.sock"))
+		if err != nil {
+			t.Fatalf("unable to run test server: %v", err)
+		}
+		defer s.Shutdown()
+		c, err := net.Dial("unixgram", addrstr)
+		if err != nil {
+			t.Fatalf("failed to dial: %v", err)
+		}
+		defer c.Close()
+		if !isPacketConn(c) {
+			t.Error("Unix datagram connection should be a packet conn")
+		}
+		if !isPacketConn(struct{ *net.UnixConn }{c.(*net.UnixConn)}) {
+			t.Error("Unix datagram connection (wrapped type) should be a packet conn")
+		}
+	})
 
-	// Unix stream
-	s, addrstr, _, err = RunLocalUnixServer(filepath.Join(t.TempDir(), "unixstream.sock"))
-	if err != nil {
-		t.Fatalf("unable to run test server: %v", err)
-	}
-	defer s.Shutdown()
-	c, err = net.Dial("unix", addrstr)
-	if err != nil {
-		t.Fatalf("failed to dial: %v", err)
-	}
-	defer c.Close()
-	if isPacketConn(c) {
-		t.Error("Unix stream connection should not be a packet conn")
-	}
-	if isPacketConn(struct{ *net.UnixConn }{c.(*net.UnixConn)}) {
-		t.Error("Unix stream connection (wrapped type) should not be a packet conn")
-	}
+	t.Run("Unix Seqpacket", func(t *testing.T) {
+		shutChan, addrstr, err := RunLocalUnixSeqPacketServer(filepath.Join(tempDir(t), "unixpacket.sock"))
+		if err != nil {
+			var sysErr syscall.Errno
+			if errors.As(err, &sysErr) && sysErr == syscall.EPROTONOSUPPORT {
```

Comment:
This can just be `errors.Is`. Is `EPROTONOSUPPORT` actually defined on all systems? Perhaps the new `errors.ErrUnsupported` is more appropriate once go1.21 lands.

Reply:
Yup, you're right. Changed